wordpress-mobile/WordPress-iOS
# Remove PostServiceRemote.trashPost (and fix rare crash in PostCategory+Lookup)

Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/23311

Removes `PostServiceRemote.trashPost`. Internally, it calls the same endpoint (`/delete`) for both self-hosted and jetpack sites.

This was the last obstacle for fully removing [`+ (RemotePost *)remotePostWithPost:(AbstractPost *)post`](https://github.com/wordpress-mobile/WordPress-iOS/blob/ac48c9bb0a419715faa51a9e3fcef263913723fd/WordPress/Classes/Services/PostHelper.m#L136), which I'll do in a separate PR.

## To test:

**Test Matrix**: self-hosted site, wp.com site

- Trash a post
- ✅ Verify that the post got trashed (is in "Trash" tab – check from the web)
- Delete a post
- ✅ Verify that the post got permanently deleted (this part is not changed)

## Regression Notes
1. Potential unintended areas of impact: Post List
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual + integration 
3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


Testing checklist:
